### ISSUE TITLE
Regression in NativeCall, seems Role related

### ISSUE THREAD

**Issue by salortiz** [rakudo/rakudo#3235]
Using a NativeCall sub in a Role fails in HEAD.

Golfed to:

```perl6
use NativeCall;
role C {
   sub calloc(size_t,size_t --> Pointer) is native(Str) { * };
   method new { say calloc(1,1) }
}
C.new()
```

Bisect log at https://gist.github.com/88a7e56a164be72cc51889e0419e7ef1

**Comment by AlexDaniel** [comment]
Clickable commits:
* 573f6a2ff2abe9ae8b6a5142b7fe5f400b651ba8
* 4df02facd08ad5c4a6031ce516909df3acf9460c
* 9ef7a7e9dd0a6baecc811faa42c1b042d8bc5452
* 1fa7ceddd424e95f1807e7ce4b34a0cc7fc6698c

**Comment by niner** [comment]
Fixed by e45bb3418d5f3964907739e916e95078e6e09995
